packages/client/src/components/Players/Player.tsx

```diff
@@ -3,31 +3,27 @@ import { CardWrapper } from 'components/Cards/CardWrapper';
 import { Deck } from 'components/Cards/Deck';
 import { playerCardWrapper } from 'components/Cards/style';
 import { InvitePlayer } from 'components/InvitePlayer';
+import { gameIO } from 'lib/socket';
 import { noop } from 'lodash';
-import { PlayerState } from 'shared';
+import { useParams } from 'react-router-dom';
+import { GameTurnEvent, PlayerState } from 'shared';
 import { Box, BoxProps, Grid } from 'theme-ui';
 import { cardKey } from 'utils/cards';
 
 type Props = {
   player: PlayerState;
   isActive: boolean;
   movingCards: string[];
-  togglePlayerActiveCard: React.Dispatch<React.SetStateAction<string | null>>;
   activePlayerCard: string | null;
 };
 
-export const Player = ({
-  player,
-  isActive,
-  movingCards,
-  togglePlayerActiveCard,
-  activePlayerCard,
-  ...rest
-}: Props & BoxProps) => {
+export const Player = ({ player, isActive, movingCards, activePlayerCard, ...rest }: Props & BoxProps) => {
+  const { roomName } = useParams<{ roomName: string }>();
```

Comment:
Isn't it better if these are passed as explicit parameters, and only the `Room` gets it from the URL?

Reply:
it doesn't matter too much for me, if possible I would avoid passing the room name, maybe we can bind roomname at the beginning and we "dispatch" an event that knows the roomName already. In any case this is the beauty of hooks, you can use them whenever you want, it doesn't harm. Why would you prefer to have them as a prop? I don't see benefits

Reply:
Because now this component is depending on the URL, right? Which is weird. Otherwise, you could put it anywhere and it would work, as long as you pass the right parameters.